new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Pod.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.assertions.impl;
+
+import java.util.ArrayList;
+import java.util.concurrent.Callable;
+
+import org.awaitility.core.ConditionFactory;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+public class Pod {
+
+  /**
+   * Check only one pod is restarted at a time in the same order as in the pods list.
+   * This assertion method needs to be called right after the domain is patched to ensure
+   * it doesn't miss any of the pods restart.
```

Comment:
I have a use case where I want to verify there I want to see the rolling restart should shutdown 2 servers simultaneously where the maxUnavailable attribute is set to 2.  In this case what will be my pod list?  Say I have 1 admin and 3 managed servers ( ms1,ms2, and ms3 ) Here  I should see ms1 and ms2 can both go to terminating state simultaneously while ms3 will be re-stated after both ms1 and ms2 are up

Reply:
According to the JIRA only one server can restart at a time.
https://jira.oraclecorp.com/jira/browse/OWLS-81478
If 2 servers can go down at the same time then how can it be a rolling restart?

Reply:
We have a field that controls how many servers can be down at the same time -- it's maxUnavailable. The default value is 1.

Reply:
This was added for customers with larger clusters who want to improve the time to roll the entire cluster.

Reply:
Yes, still, it is a rolling re-start as long as the cluster service is available. The maxUnavailable attribute is designed for this purpose. It can be any non-zero value less than the replica.  See the reference page
https://github.com/oracle/weblogic-kubernetes-operator/blob/master/docs/domains/Domain.md

Reply:
now checking the max number of pods that can restart concurrently to be not more than maxUnavailable.